drivers/serial/uart_nrfx_uarte.c

```diff
@@ -827,6 +849,11 @@ static int uarte_nrfx_tx(const struct device *dev, const uint8_t *buf,
 		(void)setup_tx_cache(dev);
 	}
 
+	if (IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME)) {
+		pm_device_runtime_get(dev);
+	}
+
+	nrf_uarte_int_enable(uarte, NRF_UARTE_INT_TXSTOPPED_MASK);
```

Comment:
I reworked it a bit. when `uart_tx()` is called we want to enable TXSTOPPED interrupt always (because it's async API). When poll_out is used then we want to enable TXSTOPPED interrupt only when PM or low power mode is used (to be able disable UARTE peripheral). I've moved enabling of TXSTOPPED to `poll_out` and removed one from `tx_start`.